ssms/drift_functions.py

```diff
@@ -238,6 +238,78 @@ def attend_drift_simple(
     return v_t
 
 
+def stimflexons_support(t, onset, coh):
+    """
+    Construct a rectangular coherence timecourse, with discrete and
+    potentially variable onset of stimulus evidence.
+
+    Arguments
+    ---------
+        t: np.ndarray
+            Timepoints within trial.
+        onset: float
+            Onset time of the coherence pulse.
+        coh: float
+            Coherence of the stimulus when 'on'.
+    Returns
+    -------
+        np.ndarray: Array of coherence values, same length as t.
+    """
+    cohs = np.zeros_like(t)
+    cohs[(t >= onset)] = coh
+    return cohs
+
+
+def ds_conflict_stimflexons_drift(
+    t: np.ndarray = np.arange(  # noqa: B008
+        0, 10, 0.001
+    ),  # TODO: #81 B008 Do not perform function call `np.arange` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable  # noqa: B008, FIX002
+    tinit: float = 0,
+    dinit: float = 0,
+    tslope: float = 1,
+    dslope: float = 1,
+    tfixedp: float = 1,
+    tcoh: float = 1.0,
+    dcoh: float = 1.0,
+    tonset: float = 0,
+    donset: float = 0,
+) -> np.ndarray:
+    """Drift function for conflict task with stimuli with potentially variable onset.
+
+    Arguments:
+    ---------
+        t: np.ndarray
+            Timepoints at which to evaluate the drift.
+            Usually np.arange() of some sort.
+        tcoh: float
+            Coherence of the target stimulus when 'on'.
+        dcoh: float
+            Coherence of the distractor stimulus when 'on'.
+        tinit: float
+            Initial condition of target drift timecourse.
+        dinit: float
+            Initial condition of distractor drift timecourse.
+        tslope: float
+            Slope parameter for target drift timecourse.
+        dslope: float
+            Slope parameter for distractor drift timecourse.
+        tfixedp: float
+            Fixed point for target drift timecourse.
+        tonset: float
+            Onset time of the target stimulus coherence.
+        donset: float
+            Onset time of the distractor stimulus coherence.
+    """
+    tcohs = stimflexons_support(t, tonset, tcoh)
```

Comment:
Consider doing something like this or removing the mutable default to make it a required argument.
```suggestion
    if t is None:
        t = np.arange(0, 20, 0.1)
    tcohs = stimflexons_support(t, tonset, tcoh)
```